test/solc/CommandLineInterface.cpp

```diff
@@ -0,0 +1,294 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+// SPDX-License-Identifier: GPL-3.0
+
+/// Unit tests for solc/CommandLineInterface.h
+
+#include <solc/CommandLineInterface.h>
+
+#include <test/Common.h>
+#include <test/FilesystemUtils.h>
+#include <test/TemporaryDirectory.h>
+#include <test/libsolidity/util/SoltestErrors.h>
+
+#include <boost/test/unit_test.hpp>
+
+#include <range/v3/range/conversion.hpp>
+#include <range/v3/view/map.hpp>
+
+#include <fstream>
+#include <map>
+#include <set>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using namespace std;
+using namespace solidity::frontend;
+using namespace solidity::test;
+
+using PathSet = set<boost::filesystem::path>;
+
+namespace
+{
+
+struct OptionsReaderAndMessages
+{
+	bool success;
+	CommandLineOptions options;
+	FileReader reader;
+	optional<string> standardJsonInput;
+	string stdoutContent;
+	string stderrContent;
+};
+
+OptionsReaderAndMessages parseCommandLineAndReadInputFiles(vector<string> const& commandLine)
+{
+	size_t argc = commandLine.size();
+	vector<char const*> argv(commandLine.size() + 1);
+
+	// argv[argc] typically contains NULL
+	argv[argc] = nullptr;
+
+	for (size_t i = 0; i < argc; ++i)
```

Comment:
```suggestion
	for (size_t const i: views::iota(argc))
```
Just a suggestion (I try to avoid mutating variables that don't have any special purpose) :)
This comes from `<ranges/views/iota.hpp>`.

Reply:
I get the idea but in this case I'd rather keep the loop as is. The current form is the bog standard idiomatic one that everyone recognizes at a glance. The one with `iota` is still simple but it would still give me a pause because it's not very common :)